modules/data.atmosphere/R/upscale_met.R

```diff
@@ -1,87 +1,85 @@
-  
-  sub_str <- substrRight(input_met, 7)
-  year <- substr(sub_str, 1, 4)
-  year <- as.numeric(year)
-  
-  # Variable names
-  var <- data.frame(CF.name <- c("air_temperature", "air_temperature_max", "air_temperature_min", 
-                                 "surface_downwelling_longwave_flux_in_air", "air_pressure", "surface_downwelling_shortwave_flux_in_air", 
-                                 "eastward_wind", "northward_wind", "specific_humidity", "precipitation_flux"), 
-                    units <- c("Kelvin", "Kelvin", "Kelvin", "W/m2", "Pascal", "W/m2", "m/s", 
-                               "m/s", "g/g", "kg/m2/s"))
-  # Reading in the data
-  met_data <- list()
+
+  loc.file = file.path(outfolder, paste("upscaled", basename(input_met), sep = "."))
+  if (file.exists(loc.file) && !isTRUE(overwrite)){
+    logger.severe("Output file", loc.file, "already exists. To replace it, set overwrite=TRUE")
+  }
+
+  met_lookup <- read.csv(system.file("/data/met.lookup.csv", package="PEcAn.data.atmosphere"),
+                         header = TRUE, stringsAsFactors = FALSE)
   tem <- ncdf4::nc_open(input_met)
   dim <- tem$dim
-  for (j in seq_along(var$CF.name)) {
-    if (exists(as.character(var$CF.name[j]), tem$var) == FALSE) {
-      met_data[[j]] <- NA
-    } else {
-      met_data[[j]] <- ncdf4::ncvar_get(tem, as.character(var$CF.name[j]))
+  met_data <- list()
+  met_units <- list()
+  for (v in names(tem$var)) {
+    if (!(v %in% met_lookup$CF_standard_name)) {
+      next
     }
+    met_data[[v]] <- ncdf4::ncvar_get(nc = tem, varid = v)
+    met_units[[v]] <- ncdf4::ncatt_get(nc = tem, varid = v, attname = "units")$value
   }
-  
+  met_data <- data.frame(met_data)
+
+  time_unit <- sub(" since.*", "", tem$dim$time$units)
+  time_base <- lubridate::parse_date_time(sub(".*since ", "", tem$dim$time$units),
+                                          orders = c("ymdHMSz", "ymdHMS", "ymd"))
+  time_data <- udunits2::ud.convert(tem$dim$time$vals, time_unit, reso_unit)
+
   lat_data <- as.numeric(ncdf4::ncvar_get(tem, "latitude"))
   lon_data <- as.numeric(ncdf4::ncvar_get(tem, "longitude"))
   ncdf4::nc_close(tem)
   
-  met_data <- data.frame(met_data)
-  colnames(met_data) <- var$CF.name
-  
-  if (lubridate::leap_year(year) == TRUE) {
-    sp <- 366
-  }
-  if (lubridate::leap_year(year) == FALSE) {
-    sp <- 365
-  }
-  reso_len <- sp * 24/reso
-  
-  step <- nrow(met_data)/reso_len
+  n_times <- diff(range(time_data)) / resolution
+  step <- round(nrow(met_data) / n_times, 0)
+  rows_used <- nrow(met_data) - (nrow(met_data) %% step)
+  n_steps <- (rows_used %/% step)
+  met_data <- met_data[1:rows_used,]
+  upscaled_time = colMeans(matrix(time_data[1:rows_used], nrow=step))
   upscale_data <- data.frame()
-  for (n in seq_along(var$CF.name)) {
-    upscale_data[1:reso_len,n] <- colMeans(matrix(met_data[[n]], nrow=step))
+  for (name in names(met_data)) {
+    upscale_data[1:n_steps,name] <- colMeans(matrix(met_data[[name]], nrow=step))
```

Comment:
per style conventions (apologies for being pedantic) .... 

1. add spaces ...

```r
upscale_data[1:n_steps,name] <- colMeans(matrix(met_data[[name]], nrow=step)) # NO!!!!
upscale_data[1:n_steps, name] <- colMeans(matrix(met_data[[name]], nrow = step)) # YES!!!!
```

2. use `seq_along` and `seq_len` for iterating with integers, e.g. 

```r
for(i in 1:nsteps){ #NO!!!!
for(i in seq_len(nsteps)){ #YES!!!
```

... bonus points for meaningful variable names: `for(step_i in seq_len(nsteps)){`